library/core/src/sync/atomic.rs

```diff
@@ -565,7 +565,8 @@ impl AtomicBool {
         since = "1.50.0",
         note = "Use `compare_exchange` or `compare_exchange_weak` instead"
     )]
-    #[cfg(target_has_atomic = "8")]
+    #[cfg_attr(not(bootstrap), cfg(target_has_atomic))]
+    #[cfg_attr(bootstrap, cfg(target_has_atomic = "8"))]
```

Comment:
Why is this necessary? If a target doesn't have 8bit atomics, it is impossible to do an AtomicBool compare_and_swap as AtomicBool is 8bit big, right? This change would add it anyway leading to either a compiler crash or miscompilation.

Reply:
Sorry, I got too aggressive with my `:%s` command. It should be fixed now - everything inside `AtomicBool` uses old annotations and I replaced them only in functions like the one below.